docs/components/checkbox-group-field/demo/base-demo.md

```diff
@@ -0,0 +1,38 @@
+```hbs template
+<Form::CheckboxGroupField
+  @label='Label'
+  @hint='Extra information about the field'
+  @name='options'
+  @value={{this.groupValue}}
+  @onChange={{this.updateValue}}
+  as |group|
+>
+  <group.CheckboxField @label='Option 1' @option='option-1' />
+  <group.CheckboxField
+    @label='Option 2'
+    @option='option-2'
```

Comment:
So the current `CheckboxField` component uses `@value` to denote if it is checked or not:

- https://github.com/CrowdStrike/ember-toucan-core/blob/main/ember-toucan-core/src/components/form/checkbox-field.hbs#L18
- https://github.com/CrowdStrike/ember-toucan-core/blob/main/ember-toucan-core/src/components/form/controls/checkbox.hbs#L6
- https://github.com/CrowdStrike/ember-toucan-core/blob/main/ember-toucan-core/src/components/form/controls/checkbox.ts#L36

So to differentiate between a checkbox group option and a checkbox "checked state", I opted for `@option` here.  We also haves some prior art for this (aka existing code) that follows the same component API pattern.

Brings up a good question though.  Maybe it'd be better to make the underlying checkbox control use a `@isChecked` instead of `@value`?  Then we can update CheckboxGroupField here to `@value` for each individual checkbox?  That may be a cleaner API.  What do you think?


```hbs
<Form::CheckboxGroupField
  @label='Label'
  @hint='Extra information about the field'
  @name='options'
  @value={{this.groupValue}}
  @onChange={{this.updateValue}}
  as |group|
>
  <group.CheckboxField @label='Option 1' @value='option-1' />
  <group.CheckboxField
    @label='Option 2'
    @value='option-2'
    @hint='Extra information about the radio'
  />
  <group.CheckboxField @label='Option 3' @option='option-3' />
  <group.CheckboxField
    @label='Option 4'
    @value='option-4'
    @hint='Extra information about the radio'
  />
</Form::CheckboxGroupField>
```


And then the checkbox-field component:

```hbs
<Form::CheckboxField
  @label='Label'
  {{! currently this is "@value" instead }}
  @isChecked={{this.value}}
  @onChange={{this.updateValue}}
/>
```

```js
import Component from '@glimmer/component';
import { action } from '@ember/object';
import { tracked } from '@glimmer/tracking';

export default class extends Component {
  @tracked value = false;

  @action
  updateValue(value) {
    this.value = value;
  }
}
```

---

If above makes sense and we want to do it, I may do that as a separate follow up PR that's focused on the component argument renaming and what not so that this PR doesn't continue to grow in scope if that's okay with you.